Mixpanel/Mixpanel.m

```diff
@@ -662,6 +662,13 @@ - (void)flushQueue:(NSMutableArray *)queue endpoint:(NSString *)endpoint
         NSString *response = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding];
         if ([response intValue] == 0) {
             MixpanelError(@"%@ %@ api rejected some items", self, endpoint);
+        } else if ([response intValue] == 1) {
+            // Success
+        } else {
+            // Another failure occurred. We expected a response of either `0` or `1`, but received something else.
+            // Maybe we were on hotel WiFi and had the request intercepted. Don't drop the data but retry again later.
+            // XXX Ideally we'd do something more intelligent, maybe by looking at `urlResponse` more closely.
+            break;
```

Comment:
@samgreen This of course assumes that the server doesn't ever send anything other than `0` or `1` (and that if it did, the request should be retried).

This is something I cannot know, but is something you'd have to verify with whoever runs that server.